crates/miden-objects/src/account/storage/slot/slot_name.rs

```diff
@@ -121,99 +94,88 @@ impl StorageSlotName {
     /// Validates a slot name.
     ///
     /// This checks that components are separated by double colons, that each component contains
-    /// only valid characters and that the name is not empty or starts or ends with a colon.
-    ///
-    /// We must check the validity of a slot name against the raw bytes of the UTF-8 string because
-    /// typical character APIs are not available in a const version. We can do this because any byte
-    /// in a UTF-8 string that is an ASCII character never represents anything other than such a
-    /// character, even though UTF-8 can contain multibyte sequences:
-    ///
-    /// > UTF-8, the object of this memo, has a one-octet encoding unit. It uses all bits of an
-    /// > octet, but has the quality of preserving the full US-ASCII range: US-ASCII characters
-    /// > are encoded in one octet having the normal US-ASCII value, and any octet with such a value
-    /// > can only stand for a US-ASCII character, and nothing else.
-    /// > https://www.rfc-editor.org/rfc/rfc3629
-    const fn validate(name: &str) -> Result<(), SlotNameError> {
-        let bytes = name.as_bytes();
-        let mut idx = 0;
+    /// only valid characters, that the name is not empty, starts or ends with a colon or starts
+    /// with an underscore.
+    fn validate(name: &str) -> Result<(), StorageSlotNameError> {
```

Comment:
nit: to be honest, I'm not sure if the new non-`const` implementation is much simpler than the `const` one. Are there any other reasons for updating it? Wondering mostly if it is worth keeping the old `const` implementation.